camunda-community-hub/camunda-cloud-helm
# Add Terratests for operate

First part of adding new helm tests. 

This PR adds:

 * a new Makefile, which makes it easy to run tests, generate golden files and develop helm charts
 * gets the new go ci github action running
 * Add golden file terratest for operate configmap
 * Add golden file terratest for operate deployment
 * Add golden files for the new tests
 * Add multiple new terratests for the operate deployment, to verify certain values are set and taken into account during rendering the templates
 * Fix certain issues with the deployment templates and default values (woho we found some issues :tada: )
 

This is the first part of many. I started with operate to show it on a less complex chart and only included two template tests, to not overwhelm the reviewer.

related https://github.com/camunda-community-hub/camunda-cloud-helm/issues/125

P.S: ~1.4 lines are go sum
